pkg/lib/hooks.js

```diff
@@ -277,7 +277,7 @@ export function useObject(create, destroy, deps, comps) {
  * }
  *
  * The component will be re-rendered whenever "proxy" emits the
- * "changed" signal.
+ * "changed" signal.  The "proxy" parameter can be null.
```

Comment:
Is this for safeguarding or what? If obj is null this hook will do nothing really.

Reply:
This is because of the "Rules of  Hooks", https://reactjs.org/docs/hooks-rules.html. Hooks need to be called in the same order for every render.  Thus, this code is not allowed because `useState` might sometimes be called first (when proxy is falsey) and sometimes second (when it is not).

```
function foo (proxy) {
    if (proxy)
        useEvent(proxy);
    const [ value, setValue ] = useState(0);
}
```

The reason for this rule is found in the implementation of hooks, and they break when we violate it.  It's not a style issue.